### ISSUE TITLE
Release 4.5.0 - Alpha 1 - DockerListener Integration testing 

### ISSUE THREAD

**Issue by fdalmaup** [wazuh/wazuh#18120]
# Docker Listener module testing

## Modules tests information
|||
|----------------------------------|------ |
| **Main release candidate issue** | #18066     |
| **Version**                      | 4.5.0 |
| **Release candidate #**          | Alpha 1   |
| **Tag**                          | [v4.5.0-alpha1](https://github.com/wazuh/wazuh/tree/v4.5.0-alpha1) |
| **Previous modules tests issue** | -    |

## Test report procedure

All test results must have one the following statuses: 
|                                  |                                                        |
|-----------------|-------------------------------------------------------------------------|
| :green_circle:  | The test passed successfully.                                           |
| :yellow_circle: | The test passed but some enhancements need to be applied to the module. |
| :red_circle:    | The test failed.                                                        |
| :white_circle:  | The test does not apply to the given bucket type or service.            |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the [Release 4.5.0 project](https://github.com/orgs/wazuh/projects/41/) for triage.

The data available in our Wazuh-dev environment for each of these supported services will be used. Additional logs may be created if they are necessary or there is insufficient data for testing.

## Test results

|Test          | Status                  | Issue ref. |
|--------------------------------|-------------------------|------------|
| Test docker listener using ossec.conf | :green_circle:  | -          |

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [x] @Selutario
- [x] @fdalmaup

**Comment by fdalmaup** [comment]
# Tests results


The [logall_json option](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall_json) must be enabled in the `ossec.conf` file before starting the tests.

## Test docker listener using ossec.conf in a manager

Apply the following configuration, restart Wazuh and check ossec.log to verify the module is running
```
<wodle name="docker-listener">
    <interval>30s</interval>
    <attempts>5</attempts>
    <run_on_start>yes</run_on_start>
    <disabled>no</disabled>
</wodle>
```

<details><summary>ossec.log output</summary>

    2023/07/28 12:46:53 wazuh-modulesd:docker-listener[3063] wm_docker.c:46 at wm_docker_main(): INFO: Module docker-listener started.
    2023/07/28 12:46:53 wazuh-modulesd:docker-listener[3063] wm_docker.c:59 at wm_docker_main(): INFO: Starting to listening Docker events.
    2023/07/28 12:46:53 wazuh-modulesd:docker-listener[3063] wm_docker.c:63 at wm_docker_main(): DEBUG: Launching command 'wodles/docker/DockerListener'

</details>


Run a container with no cache available
```
docker run ubuntu
```

<details><summary>Archives.json output</summary>

    {"timestamp":"2023-07-28T15:47:35.140+0000","rule":{"level":3,"description":"Docker: Image or repository ubuntu pulled","id":"87932","firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559255.746164","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"pull\", \"id\": \"ubuntu:latest\", \"Type\": \"image\", \"Action\": \"pull\", \"Actor\": {\"ID\": \"ubuntu:latest\", \"Attributes\": {\"name\": \"ubuntu\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1690559255, \"timeNano\": 1690559255139807038}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"pull","id":"ubuntu:latest","Type":"image","Action":"pull","Actor":{"ID":"ubuntu:latest","Attributes":{"name":"ubuntu","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1690559255","timeNano":"1690559255139806976.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-07-28T15:47:35.185+0000","rule":{"level":3,"description":"Docker: Container serene_chatterjee created","id":"87901","firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559255.747134","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"create\", \"id\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"create\", \"Actor\": {\"ID\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"serene_chatterjee\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1690559255, \"timeNano\": 1690559255185209800}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"create","id":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","from":"ubuntu","Type":"container","Action":"create","Actor":{"ID":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","Attributes":{"image":"ubuntu","name":"serene_chatterjee","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1690559255","timeNano":"1690559255185209856.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-07-28T15:47:35.186+0000","rule":{"level":3,"description":"Docker: Attached local standard input, output, and error streams to container serene_chatterjee","id":"87922","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559255.748443","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"attach\", \"id\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"attach\", \"Actor\": {\"ID\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"serene_chatterjee\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1690559255, \"timeNano\": 1690559255186418798}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"attach","id":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","from":"ubuntu","Type":"container","Action":"attach","Actor":{"ID":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","Attributes":{"image":"ubuntu","name":"serene_chatterjee","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1690559255","timeNano":"1690559255186418688.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-07-28T15:47:35.207+0000","rule":{"level":3,"description":"Docker: Network bridge connected","id":"87928","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559255.749725","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"connect\", \"Actor\": {\"ID\": \"175a655cb766eee2fb96644c1d50db76f101dd59a1c5da0f28d2bb8fad60fef4\", \"Attributes\": {\"container\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1690559255, \"timeNano\": 1690559255206872219}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"connect","Actor":{"ID":"175a655cb766eee2fb96644c1d50db76f101dd59a1c5da0f28d2bb8fad60fef4","Attributes":{"container":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","name":"bridge","type":"bridge"}},"scope":"local","time":"1690559255","timeNano":"1690559255206872320.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-07-28T15:47:35.434+0000","rule":{"level":3,"description":"Docker: Container serene_chatterjee started","id":"87903","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559255.750647","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"start\", \"id\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"start\", \"Actor\": {\"ID\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"serene_chatterjee\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1690559255, \"timeNano\": 1690559255434572326}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"start","id":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","from":"ubuntu","Type":"container","Action":"start","Actor":{"ID":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","Attributes":{"image":"ubuntu","name":"serene_chatterjee","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1690559255","timeNano":"1690559255434572288.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-07-28T15:47:35.629+0000","rule":{"level":4,"description":"Docker: Network bridge disconnected","id":"87929","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559255.751873","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"disconnect\", \"Actor\": {\"ID\": \"175a655cb766eee2fb96644c1d50db76f101dd59a1c5da0f28d2bb8fad60fef4\", \"Attributes\": {\"container\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1690559255, \"timeNano\": 1690559255628599416}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"disconnect","Actor":{"ID":"175a655cb766eee2fb96644c1d50db76f101dd59a1c5da0f28d2bb8fad60fef4","Attributes":{"container":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","name":"bridge","type":"bridge"}},"scope":"local","time":"1690559255","timeNano":"1690559255628599296.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-07-28T15:47:35.635+0000","rule":{"level":7,"description":"Docker: Container serene_chatterjee received the action: die","id":"87924","firedtimes":1,"mail":false,"groups":["docker"],"gdpr":["IV_32.2"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559255.752804","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"die\", \"id\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"die\", \"Actor\": {\"ID\": \"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df\", \"Attributes\": {\"execDuration\": \"0\", \"exitCode\": \"0\", \"image\": \"ubuntu\", \"name\": \"serene_chatterjee\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1690559255, \"timeNano\": 1690559255635301691}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"die","id":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","from":"ubuntu","Type":"container","Action":"die","Actor":{"ID":"0b680931c2f383ea7914d432048e56516cf8310b07abc2d08d6c0ee8378bd6df","Attributes":{"execDuration":"0","exitCode":"0","image":"ubuntu","name":"serene_chatterjee","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1690559255","timeNano":"1690559255635301632.000000"}},"location":"Wazuh-Docker"}

</details>


Remove the image

<details><summary>Archives.json output</summary>

    {"timestamp":"2023-07-28T15:50:29.858+0000","rule":{"level":3,"description":"Docker: Image sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb untagged","id":"87919","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559429.754166","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"untag\", \"id\": \"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb\", \"Type\": \"image\", \"Action\": \"untag\", \"Actor\": {\"ID\": \"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb\", \"Attributes\": {\"name\": \"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1690559429, \"timeNano\": 1690559429857823351}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"untag","id":"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb","Type":"image","Action":"untag","Actor":{"ID":"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb","Attributes":{"name":"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1690559429","timeNano":"1690559429857823232.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-07-28T15:50:29.859+0000","rule":{"level":7,"description":"Docker: Container sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb deleted","id":"87921","mitre":{"id":["T1070.004","T1561.001"],"tactic":["Defense Evasion","Impact"],"technique":["File Deletion","Disk Content Wipe"]},"firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7","11.5"],"hipaa":["164.312.b","164.312.c.1","164.312.c.2"],"nist_800_53":["AU.14","SI.7"],"tsc":["CC6.8","CC7.2","CC7.3","PI1.4","PI1.5","CC6.1"]},"agent":{"id":"000","name":"487357a9f0b1"},"manager":{"name":"487357a9f0b1"},"id":"1690559429.754166","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"delete\", \"id\": \"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb\", \"Type\": \"image\", \"Action\": \"delete\", \"Actor\": {\"ID\": \"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb\", \"Attributes\": {\"name\": \"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb\"}}, \"scope\": \"local\", \"time\": 1690559429, \"timeNano\": 1690559429859609520}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"delete","id":"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb","Type":"image","Action":"delete","Actor":{"ID":"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb","Attributes":{"name":"sha256:5a81c4b8502e4979e75bd8f91343b95b0d695ab67f241dbed0d1530a35bde1eb"}},"scope":"local","time":"1690559429","timeNano":"1690559429859609600.000000"}},"location":"Wazuh-Docker"}

</details>


### Conclusions

The module worked as expected

**Comment by davidjiglesias** [comment]
LGTM!
